File: programming/temporal_overlap_probability/python/temporal_overlap_probability.py
import numpy as np

def get_annual_cost():
    cost_per_downtime = 1000

    # let's simulate 10000 starting minutes of the 2 jobs
    n = 10000
    task1 = np.random.randint(0, 300, size=n)
    task2 = np.random.randint(0, 300, size=n)

    # let's find the overlaps and calculate the probability of the overlaps
    overlaps = sum(abs(task1-task2) <= 60)
    prob_overlap = overlaps / n

    # return the annual cost estimate based on the probability
    return prob_overlap * 365 * cost_per_downtime

File: programming/temporal_overlap_probability/R/temporal_overlap_probability.R
get_annual_cost <- function(cost_per_downtime, job_length_in_mins, time_window_in_mins) {
    # let's simulate 10000 starting minutes of the 2 jobs
    n <- 10000
    task1 <- sample.int(time_window_in_mins, size=n, replace=TRUE)
    task2 <- sample.int(time_window_in_mins, size=n, replace=TRUE)
    overlaps <- abs(task1 - task2) <= job_length_in_mins
    overlap_prob <- sum(overlaps) / n

    overlap_prob * 365 * cost_per_downtime
}